### PR TITLE
planner: hide cost model factors from users

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #36243

Problem Summary: hide cost model factors from users

### What is changed and how it works?

Previously, we implemented cost factors as exposed SQL variables, which brings us some troubles:
1. Cost model factors are very internal concepts of the optimizer so exposing them may confuse the users; 
2. These factors can work together well only when they are carefully calibrated and aligned with each other, so implementing them as multiple SQL variables gives users the opportunity to break it, which brings some risks;
3. These variables are written into the storage layer, which brings some compatibility issues when updating the cost model.

So after discussing with some team members and investigating some other DBs' Implementation, we decided to hide cost model factors from users in Cost Model Ver2. (Cost Model Ver1 remains unchanged)

Cost model factors are very internal concepts of the optimizer. 

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by chrysan — APPROVED** [review]

File: planner/core/plan_cost.go [hunk 1]
```diff
@@ -1033,13 +1030,6 @@ func (p *PhysicalStreamAgg) GetCost(inputRows float64, isRoot, isMPP bool, costF
 	sessVars := p.ctx.GetSessionVars()
 	if isRoot {
 		cpuCost = inputRows * sessVars.GetCPUFactor() * aggFuncFactor
-	} else if isMPP {
-		if p.ctx.GetSessionVars().CostModelVersion == modelVer2 {
-			// use the dedicated CPU factor for TiFlash on modelVer2
-			cpuCost = inputRows * sessVars.GetTiFlashCPUFactor() * aggFuncFactor
-		} else {
-			cpuCost = inputRows * sessVars.GetCopCPUFactor() * aggFuncFactor
```
Comment (time-and-fate) [inline planner/core/plan_cost.go:1041]: Why is this removed?

File: planner/core/plan_cost.go [hunk 2]
```diff
@@ -1082,13 +1072,6 @@ func (p *PhysicalHashAgg) GetCost(inputRows float64, isRoot, isMPP bool, costFla
 			// Cost of additional goroutines.
 			cpuCost += (con + 1) * sessVars.GetConcurrencyFactor()
 		}
-	} else if isMPP {
-		if p.ctx.GetSessionVars().CostModelVersion == modelVer2 {
-			// use the dedicated CPU factor for TiFlash on modelVer2
-			cpuCost = inputRows * sessVars.GetTiFlashCPUFactor() * aggFuncFactor
-		} else {
-			cpuCost = inputRows * sessVars.GetCopCPUFactor() * aggFuncFactor
```
Comment (time-and-fate) [inline planner/core/plan_cost.go:1090]: ditto

**Review by XuHuaiyu — APPROVED** [review]

File: planner/core/plan_cost_ver2.go [hunk 3]
```diff
@@ -541,71 +541,119 @@ func (p *BatchPointGetPlan) getPlanCostVer2(taskType property.TaskType, option *
...
+	TiDB2FlashNet: costVer2Factor{"tidb_flash_net_factor", 4},
+	TiFlashMPPNet: costVer2Factor{"tiflash_mpp_net_factor", 4},
+	TiDBMem:       costVer2Factor{"tidb_mem_factor", 1},
+	TiKVMem:       costVer2Factor{"tikv_mem_factor", 1},
+	TiFlashMem:    costVer2Factor{"tiflash_mem_factor", 1},
+	TiDBDisk:      costVer2Factor{"tidb_disk_factor", 1000},
+	TiDBRequest:   costVer2Factor{"tidb_request_factor", 9500000},
+}
+
+func getTaskCPUFactor(p PhysicalPlan, taskType property.TaskType) costVer2Factor {
```
Comment (time-and-fate) [inline planner/core/plan_cost_ver2.go:631]: Do we need to name it and the methods below something like `getTaskCPUFactor`? There are methods with very similar names in `plan_cost.go`.

File: planner/core/plan_cost.go (hunk 1 above)
Comment (qw4990) [reply planner/core/plan_cost.go:1041]: Since it cannot be modelVer2 here.

File: planner/core/plan_cost_ver2.go (hunk 3 above)
Comment (qw4990) [reply planner/core/plan_cost_ver2.go]: OK, I renamed them to `getXXXFactorVer2` to indicate they are only used by model ver2.

File: planner/core/plan_cost_ver2.go [hunk 4]
```diff
@@ -617,40 +665,30 @@ func getTaskScanFactor(p PhysicalPlan, taskType property.TaskType) float64 {
...
+			return defaultVer2Factors.TiDB2FlashNet
 		}
-		return getTableInfo(x.Children()[0])
 	}
+	return defaultVer2Factors.TiDB2KVNet
+}
+
+func getTaskSeekFactorVer2(p PhysicalPlan, _ property.TaskType) costVer2Factor {
+	return defaultVer2Factors.TiDBRequest
 }
```
Comment (time-and-fate) [inline planner/core/plan_cost_ver2.go:692]: I think the logic changed here.
1. It doesn't recursively find `TableInfo` in the children now.
2. It doesn't return 0 for the temporary table now.

File: planner/core/plan_cost.go (hunk 1 above)
Comment (time-and-fate) [reply planner/core/plan_cost.go:1041]: But the `else` branch means it's version 1 here.

Comment (time-and-fate) [reply planner/core/plan_cost.go:1041]: Got it. I mistook it. It's actually the same as before.
